lib/src/handler_client.dart

```diff
@@ -9,15 +9,26 @@ import 'handler.dart';
 import 'request.dart';
 import 'response.dart';
 
+/// A [Handler] based HTTP client.
+///
+/// The [HandlerClient] allows composition of a [Client] within a larger
+/// application.
 class HandlerClient extends BaseClient {
   final Handler _handler;
   final void Function() _close;
 
+  /// Creates a new client using the [_handler] and [onClose] functions.
   HandlerClient(this._handler, void onClose())
       : _close = onClose;
 
+  /// Sends an HTTP request and asynchronously returns the response.
+  ///
+  /// The [Handler] function will be called when this method is invoked.
```

Comment:
I don't think this line is necessary, especially since you can't directly refer to the values. Same for the similar line in `close()`.